### DISCUSSION TITLE
Accepting only one version of an entry

### DISCUSSION THREAD

**Discussion by Spedwards** [Flexget/Flexget#3642] [Q&A]
I've recently switched to using FlexGet for downloading torrents from an RSS feed. I was previously using the RSS plugin within rTorrent/ruTorrent but it had limitations.

I'm using modified versions of the `rss` and `rtorrent` plugins as input and output respectively, as well as a fully custom plugin I'm using to filter the feed and only download entries I actually want.

Occasionally, there'll be a different version of the same entry (codec, resolution, etc.) and currently FlexGet will grab both versions. For example, here are the titles of two different items in the feed:

 - Just a random item [1080p]
 - Just a random item [1080p, x265]

Unfortunately, they don't all follow a standard but matching pairs (triplets, etc.) will often follow the same style, meaning it's unlikely to have two files like this:

 - Just a random item [1080p]
 - Just a random item - 1080p - x265

What I'd like to be able to do is to reject non-x265 items if the x265 item has already been accepted (or rather, just seen). x265 items should still get accepted if they come later. Is there a way I can do this within my filtering plugin? Can I get a list of seen entries and perform a check?

**Comment by lostb1t** [comment]
If its in the same task run you can use https://flexget.com/Plugins/best_quality

if you want to filter out already downloaded files then you can populate a entry/movie list with the filesystem plugin and crossmatch/match it from the task you want filtered 

there is also the upgrade plugin: https://flexget.com/Plugins/upgrade

**Reply by Spedwards** [reply]
I already basically have quality built into my custom filter. It has a global quality config option as well as optional quality settings for each item.

The upgrade plugin looks like it _might_ work, though I'll have to adapt it for my needs of course. I've only skimmed through the code so I can't say for certain though.

I'll look closer at it tomorrow and see if I can get it adapted to my purposes.

**Reply by crawln45** [reply]
If you only want to accept x265 codec it might be in the qualities, I'd have to check myself as I haven't looked at them in a bit. If it is you can specify encoding as well as resolution, audio codec, ext. Also you may have some issues if the movie in this case is not removed from the list after it had been downloaded. But usually the seen plugin would detect that so make sure you haven't disabled it as well. There are a few checks that are done to ensure this.
